Yuneec_SDK_iOS/YNCSDKCamera.mm

```diff
@@ -26,6 +26,93 @@ void receive_camera_result(YNCCameraCompletion completion, Camera::Result result
+    if (completion) {
+        YNCCameraMode tmpMode = YNCCameraMode::YNCCameraModeUnknown;
+        NSError *error = nullptr;
+        if (result != Camera::Result::SUCCESS) {
+            NSString *message = [NSString stringWithFormat:@"%s", Camera::result_str(result)];
+            error = [[NSError alloc] initWithDomain:@"Camera"
+                                               code:(int)result
+                                           userInfo:@{@"message": message}];
+        completion(tmpMode, error);
+        }
+        else {
+            switch (mode) {
+                case Camera::Mode::PHOTO:
+                    tmpMode = YNCCameraModePhoto;
+                    break;
+                case Camera::Mode::PHOTO_SURVEY:
+                    tmpMode = YNCCameraModePhotoSurvey;
+                    break;
+                case Camera::Mode::VIDEO:
+                    tmpMode = YNCCameraModeVideo;
+                    break;
+                default:
+                    tmpMode = YNCCameraModeUnknown;
+                    break;
+            }
+        }
+        completion(tmpMode, error);
+    }
+}
+
+//MARK: receive camera status result
+void receive_camera_status_result(YNCCameraStatusCompletion completion, Camera::Result result, Camera::Status status) {
+    if (completion) {
+        NSError *error = nullptr;
+        if (result != Camera::Result::SUCCESS) {
+            NSString *message = [NSString stringWithFormat:@"%s", Camera::result_str(result)];
+            error = [[NSError alloc] initWithDomain:@"Camera"
+                                               code:(int)result
+                                           userInfo:@{@"message": message}];
+            completion(nil, error);
+        }
+        else {
+            YNCCameraStatus *tmpStatus = [YNCCameraStatus new];
+            tmpStatus.photoIntervalOn = status.photo_interval_on;
+            tmpStatus.videoOn = status.video_on;
+            tmpStatus.availableStorageMib = status.available_storage_mib;
+            tmpStatus.totalStorageMib = status.total_storage_mib;
+            tmpStatus.usedStorageMib = status.used_storage_mib;
+            switch (status.storage_status) {
+                case Camera::Status::StorageStatus::FORMATTED:
+                    tmpStatus.storageOk = true;
+                    break;
+                case Camera::Status::StorageStatus::UNFORMATTED:
+                    tmpStatus.storageOk = true;
+                    break;
+                case Camera::Status::StorageStatus::NOT_AVAILABLE:
+                    tmpStatus.storageOk = false;
+                    break;
+                default:
+                    break;
+            }
+            completion(tmpStatus, error);
+        }
+    }
+}
+
+Camera::Mode getCameraModeEnum(YNCCameraMode mode) {
+    Camera::Mode cameraMode;
+    switch (mode) {
+        case YNCCameraModePhoto:
+            cameraMode = Camera::Mode::PHOTO;
```

Comment:
If you want to simplify this function, you could just directly do `return Camera::Mode::Photo;` instead of `break`ing.